sklearn/metrics/tests/test_ranking.py

```diff
@@ -377,27 +379,30 @@ def test_roc_curve_toydata():
     )
     with pytest.warns(UndefinedMetricWarning, match=expected_message):
         tpr, fpr, _ = roc_curve(y_true, y_score)
-
-    with pytest.raises(ValueError):
+    # should raise a warning since only one class is present
+    expected_message = (
+        "Only one class present in y_true. ROC AUC score is not defined in that case."
+    )
```

Comment:
This seems to be redundant. Can we simply use the expected_message defined above?
```suggestion
    # should raise a warning since only one class is present
```

Reply:
But these are two different messages for two different edge cases. Should I use only one of them everywhere?

Reply:
Isn't the content of the message the same? I think if the content is the same even if we are testing two different cases we can reuse the same message that we declared initially.

Reply:
This is the same use case but different functions.

for inputs
```
y_true = [0, 0]
y_score = [0.25, 0.75]
```
we expect warning
```
expected_message = (
    "No positive samples in y_true, true positive value should be meaningless"
)
```
from the call `roc_curve(y_true, y_score)`

and we expect warning
```
expected_message = (
    "Only one class is present in y_true. "
    "ROC AUC score is not defined in that case."
)
``` 
from the call `roc_auc_score(y_true, y_score)`

Thus we have to rewrite the expected_message.

Content is the same but in `roc_auc_score` we check if number of unique values in `y_true` is less than 2 right at the very beginning, so there is no fpr/tpr calculations if that's the case. So we cannot raise the same errors in both of this cases.

Reply:
Maybe better way would be split this test function into two: one for `roc_auc_score` and one for `roc_curve`?

Reply:
I think let's keep it the way it is. Thanks for trying it out.